aten/src/ATen/native/cuda/CuFFTPlanCache.h

```diff
@@ -280,65 +352,38 @@ class CuFFTConfig {
         batch, &ws_size_t, exec_type));
 #endif
     } else {
-      // set idist (stride at batch dim)
-      // set base_istride (stride at innermost dim of signal)
-      long long int idist, base_istride;
-      if (clone_input) {
-        idist = at::prod_intlist(input.sizes().slice(1, signal_ndim));
-        base_istride = 1;
-      } else if (complex_input) {
-        idist = input.stride(0) >> 1;
-        base_istride = input.stride(signal_ndim) >> 1;
-      } else {
-        idist = input.stride(0);
-        base_istride = input.stride(signal_ndim);
-      }
-      // Even if batch dimension is one and idist (stride(0)) doesn't matter,
-      // cuFFT errors if idist = 0. This is hack to make it succeed.
-      if (idist == 0 && batch == 1) {
-        idist = 1;
-      }
-
-      // set odist, onembed, base_ostride
 #ifdef __HIP_PLATFORM_HCC__
-      int odist = at::prod_intlist(output_sizes.slice(1, signal_ndim));
-      std::vector<int> onembed(output_sizes.data() + 1, output_sizes.data() + signal_ndim + 1);
-      int base_ostride = 1;
-
-      int istride = base_istride;
-      int iidist = idist;
       CUFFT_CHECK(hipfftMakePlanMany(plan(), signal_ndim, signal_sizes.data(),
-        inembed.data(), istride, iidist,
-        onembed.data(), base_ostride, odist,
+        in_layout.embed.data(), in_layout.stride, in_layout.dist,
+        out_layout.embed.data(), out_layout.stride, out_layout.dist,
         exec_type, batch, &ws_size_t));
 #else
-      long long int odist = at::prod_intlist(output_sizes.slice(1, signal_ndim));
-      std::vector<long long int> onembed(output_sizes.data() + 1, output_sizes.data() + signal_ndim + 1);
-      long long int base_ostride = 1;
-
       CUFFT_CHECK(cufftXtMakePlanMany(plan(), signal_ndim, signal_sizes.data(),
-            inembed.data(), base_istride, idist, itype,
-            onembed.data(), base_ostride, odist, otype,
+            in_layout.embed.data(), in_layout.stride, in_layout.dist, itype,
+            out_layout.embed.data(), out_layout.stride, out_layout.dist, otype,
             batch, &ws_size_t, exec_type));
 #endif
-      }
+    }
     ws_size = static_cast<int64_t>(ws_size_t);
   }
 
 #ifdef __HIP_PLATFORM_HCC__
-  cufftHandle &plan() const { return *plan_ptr.get(); }
+  cufftHandle &plan() const { return plan_ptr.get(); }
 #else
-  const cufftHandle &plan() const { return *plan_ptr.get(); }
+  const cufftHandle &plan() const { return plan_ptr.get(); }
 #endif
 
+  CuFFTTransformType transform_type() const { return fft_type_; }
+  ScalarType data_type() const { return value_type_; }
   bool should_clone_input() const { return clone_input; }
-
   int64_t workspace_size() const { return ws_size; }
 
 private:
-  std::unique_ptr<cufftHandle, CuFFTHandleDeleter> plan_ptr;
+  mutable CuFFTHandle plan_ptr;
```

Comment:
Are there are alternatives to making this mutable?

Reply:
The mutable was to deal with the the `#ifdef __HIP_PLATFORM_HCC__` above that takes a mutable reference to the handle, but I was able to compile on ROCm without the `ifdef` so lets see if CI comes back clean without it.